ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -718,8 +738,17 @@ private Page<NamedBlobRecord> run_list_v2(String accountName, String containerNa
 
           entries.add(
               new NamedBlobRecord(accountName, containerName, blobName, blobId, timestampToMs(deletionTime), version,
-                  blobSize, timestampToMs(modifiedTime)));
+                  blobSize, timestampToMs(modifiedTime), false));
         }
+
+        if (groupDirectories) {
+          // Add the directories to the result
+          entries.addAll(directories.stream()
+              .map(directory -> new NamedBlobRecord(accountName, containerName, directory, null, Utils.Infinite_Time, 0,
```

Comment:
This is not right, when we are returning a common prefix, it should include the prefix as well. if we have "abc/def/ghi", and the prefix is "abc/", and the common prefix should be "abc/def/", not jsut "def/".

Reply:
Corrected it..